shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -143,6 +143,17 @@ class JSBackend {
     case StrLit(value) => JSLit(JSLit.makeStringLiteral(value))
     // `Asc(x, ty)` <== `x: Type`
     case Asc(trm, _) => translateTerm(trm)
+    // `c with { x = "hi"; y = 2 }` ==> `Object.assign(c, { x: "hi", y: 2 })`
```

Comment:
As discussed in the meeting, this is not sound as it should create a new object, not update the old one in place.